src/main/org/deidentifier/arx/io/ImportAdapterJDBC.java

```diff
@@ -1,309 +1,419 @@
+     * <p>
+     * The first row contains the name of the columns. Depending upon whether
+     * the name of the column has been assigned explicitly, this is either the
+     * value of the table itself, the value defined by the user.
+     */
+    private boolean headerReturned;
+
+    /**
+     * Number of rows that need to be processed in total.
+     *
+     * @see {@link #getProgress()}
+     */
+    private int totalRows;
+
+    /**
+     * Creates a new instance of this object with given configuration.
+     *
+     * @param config {@link #config}
+     * @throws IOException In case of communication errors with JDBC
+     * @todo Fix IOException
+     */
+    protected ImportAdapterJDBC(ImportConfigurationJDBC config) throws IOException {
+
+        super(config);
+        this.config = config;
+
+        /* Preparation work */
+        indexes = getIndexesToImport();
+        dataTypes = getColumnDatatypes();
+
+        try {
+
+            /* Used to keep track of progress */
+            statement = config.getConnection().createStatement();
+
+            String tableName = config.getTable();
+            String schemaName = config.getSchemaName();
+            String dbName = config.getDbName();
+            DatabaseMetaData meta = config.getConnection().getMetaData();
+            String productName = meta.getDatabaseProductName().toLowerCase();
+
+            String afterTableName = "";
+
+            String databaseType = "";
+
+            if (productName.contains("oracle")) {
+                databaseType = "oracle";
+            } else if (productName.contains("db2")) {
+                databaseType = "db2";
+            } else if (productName.startsWith("microsoft sql server")) {
+                databaseType = "sqlserver";
+            } else if (productName.contains("mysql")) {
+                databaseType = "mysql";
+            } else {
+                databaseType = "other";
+            }
+
+            switch (databaseType) {
+                case "mysql":
+                    afterTableName = "`" + schemaName + "`.`" + tableName + "`";
+                    break;
+                case "oracle":
+                    afterTableName = "\"" + schemaName + "\".\"" + tableName + "\"";
+                    break;
+                case "db2":
+                    afterTableName = "\"" + schemaName + "\".\"" + tableName + "\"";
+                    break;
+                case "sqlserver":
+                    afterTableName = "[" + dbName + "].[" + schemaName + "].[" + tableName + "]";
+                    break;
+                default:
+                    afterTableName = tableName;
+                    break;
+            }
+
+
+            statement.execute("SELECT COUNT(*) FROM " + afterTableName);
+            resultSet = statement.getResultSet();
+
+            if (resultSet.next()) {
+
+                totalRows = resultSet.getInt(1);
+
+                if (config.getMaxRows() != 0) {
```

Comment:
See also above. Why is totalRows needed? Wouldn't config.getMaxRows() be sufficient?